server/src/main/java/org/opensearch/rest/pagination/IndexBasedPaginationStrategy.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.common.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This strategy can be used by the Rest APIs wanting to paginate the responses based on Indices.
+ * The strategy considers create timestamps of indices as the keys to iterate over pages.
+ *
+ * @opensearch.internal
+ */
+public class IndexBasedPaginationStrategy implements PaginationStrategy<String> {
+
+    private final IndexStrategyPageToken nextToken;
```

Comment:
Could we use PageRequestMetadata type itself insted of IndexStrategyPageToken?

Reply:
Updated the strategy to interact with only PaginatedQueryRequest and PaginatedQueryResponse. Removed the PageToken interface which was causing confusions and wasn't really adding much value.